bundles/org.openhab.binding.bluetooth.dbusbluez/src/main/java/org/openhab/binding/bluetooth/dbusbluez/handler/DBusBlueZBridgeHandler.java

```diff
@@ -0,0 +1,343 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.dbusbluez.handler;
+
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.freedesktop.dbus.exceptions.DBusException;
+import org.openhab.binding.bluetooth.AbstractBluetoothBridgeHandler;
+import org.openhab.binding.bluetooth.BluetoothAddress;
+import org.openhab.binding.bluetooth.dbusbluez.DBusBlueZBluetoothDevice;
+import org.openhab.binding.bluetooth.dbusbluez.internal.DBusBlueZUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.hypfvieh.bluetooth.DeviceManager;
+
+/**
+ * The {@link DBusBlueZBridgeHandler} is responsible for talking to the BlueZ stack, using DBus Unix Socket.
+ * This Binding does not use any JNI.
+ * It provides a private interface for {@link DBusBlueZBluetoothDevice}s to access the stack and provides top
+ * level adaptor functionality for scanning and arbitration.
+ *
+ * @author Benjamin Lafois - Initial contribution and API
+ */
+@NonNullByDefault
+public class DBusBlueZBridgeHandler extends AbstractBluetoothBridgeHandler<DBusBlueZBluetoothDevice> {
+
+    private final Logger logger = LoggerFactory.getLogger(DBusBlueZBridgeHandler.class);
+
+    // ADAPTER from BlueZ-DBus Library
+    private @NonNullByDefault({}) com.github.hypfvieh.bluetooth.wrapper.BluetoothAdapter adapter;
+    private @NonNullByDefault({}) com.github.hypfvieh.bluetooth.DeviceManager deviceManager;
+
+    // Our BT address
+    private @NonNullByDefault({}) BluetoothAddress adapterAddress;
+
+    private DBusBlueZPropertiesChangedHandler propertiesChangedHandler = new DBusBlueZPropertiesChangedHandler();
+
+    private final ReentrantLock lockDiscoveryJob = new ReentrantLock();
+
+    private @NonNullByDefault({}) ScheduledFuture<?> discoveryJob;
```

Comment:
The typical pattern for bindings is to make ScheduledFutures nullable.
```suggestion
    private @Nullable ScheduledFuture<?> discoveryJob;
```